src/main/java/io/github/ryotan/code/constraint/CodeValue.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * These codes are licensed under CC0.
+ *
+ * http://creativecommons.org/publicdomain/zero/1.0/deed
+ * http://creativecommons.org/publicdomain/zero/1.0/deed.ja
+ */
+
+package io.github.ryotan.code.constraint;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.function.Predicate;
+import javax.validation.Constraint;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import io.github.ryotan.code.Code;
+import io.github.ryotan.code.CodeEnum;
+import io.github.ryotan.code.CodeEnum.Filters;
+import io.github.ryotan.code.util.CodeEnumReflectionUtil;
+
+@Target({METHOD, FIELD, PARAMETER})
+@Retention(RUNTIME)
+@Documented
+@Constraint(validatedBy = {CodeValue.CodeValueValidator.class, CodeValue.CodeEnumValidator.class})
+public @interface CodeValue {
+
+    String message() default "{io.github.ryotan.code.constraint.CodeValue.message}";
+
+    Class<?>[] groups() default {};
+
+    Class<?>[] payload() default {};
+
+    Class<? extends CodeEnum<?>> value();
+
+    String[] filters() default {};
+
+    abstract class CodeEnumValidatorSupport<T> implements ConstraintValidator<CodeValue, T> {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Type Javadoc comment is missing an null tag.](https://www.codacy.com/app/ryotan/code-enum-support/file/3194383390/issues/source?bid=3375378&fileBranchId=3375797#l44)